app/models/assessment_section.rb

```diff
@@ -23,6 +23,7 @@
 #
 class AssessmentSection < ApplicationRecord
   belongs_to :assessment
+  has_many :assessment_section_failure_reasons
```

Comment:
I wonder if we should rename `failure_reasons` to `available_failure_reason_keys` at some point.

Reply:
Yeah. We should have a think about the naming. I'm still not really into 'failure_` either but that's probably stuck now. I'll update them to something in the last PR when the old implementation is gone which will free up `selected_failure_reasons`.